rpe/resources/gcp.py

```diff
@@ -234,6 +237,52 @@ def _update_request_args(self, body):
             'project': self.resource_data['project_id']
         }
 
+class GcpComputeSubnetwork(GoogleAPIResource):
+
+    service_name = "compute"
+    resource_path = "subnetworks"
+    version = "v1"
+    update_method = "patch"
+
+    def _get_request_args(self):
+        return {
+            'project': self.resource_data['project_id'],
+            'region': self.resource_data['resource_location'],
+            'subnetwork': self.resource_data['resource_name']
+        }
+
+    def _update_request_args(self, body):
+        return {
+            'project': self.resource_data['project_id'],
+            'region': self.resource_data['resource_location'],
+            'subnetwork': self.resource_data['resource_name'],
+            'body': body
+        }
+
+class GcpComputeSubnetworkPrivateip(GcpComputeSubnetwork):
+
+    resource_property = 'privateip'
+    update_method = "setPrivateIpGoogleAccess"
```

Comment:
To avoid having separate 'resources' for each property, can we use the subnetwork `patch` method instead? The resource properties were more to handle things outside of a resources metadata...specifically IAM policy at the time. This is a workable solution, but if we can avoid the complexity that would be preferred.

Reply:
As per [docs](https://developers.google.com/resources/api-libraries/documentation/compute/v1/python/latest/compute_v1.subnetworks.html#patch), _privateIpGoogleAccess_ can be only set during creation, or updated using _setPrivateIpGoogleAccess_ method. Including it in _patch_ request returns immutable field error.

Reply:
That makes sense, and this is a well designed solution. I'm worried about the increase in `resources` as we run into more of these situations (and im aware of a few on the GKE and Compute APIs) so I'd like to find a cleaner solution. 

I'm working on a new remedation spec (#22 ) and I'm hoping to get that merged soon. Using the new spec, the OPA policy will be able to return the name of a method to call, and the payload. This will eliminate the need for additional resources for this case. As an example remediation for this may look like this:

```
remediate = {
  "_remediation_spec": "v2beta1",
  "steps": [
    enable_private_google_access
  ]
}

enable_private_google_access = {
    "method": "setPrivateIpGoogleAccess",
    "params": {
        "subnetwork": input.name,
        "region": input.region,
        "project": selfLinkParts[1],
        "body":  {
            "privateIpGoogleAccess": true
        }
    }
}

# break out the selfLink so we can extract the project
selfLinkParts = split(input.selfLink,"/")
```

Reply:
The PR mentioned above was approved. Please use the new remediation spec, rather than create separate classes for resource properties such as privateGoogleAccess. Thank you for catching this scenario! Sorry for asking you to redo some of the work here, but in the end I think it will be a better solution.

Reply:
New remediation spec is now used for privateGoogleAccess, changes are committed.